Installation on macOS fails to find brew libudunits

Console session of an installation attempt of the dev package on a pretty vanilla macOS setup:

```
tomasz@tomaszkalinows-WQVX units % brew install udunits 
Warning: udunits 2.2.28 is already installed and up-to-date.
To reinstall 2.2.28, run:
  brew reinstall udunits
tomasz@tomaszkalinows-WQVX units % Rscript -e 'remotes::install_local()'

── R CMD build ─────────────────────────────────────────────────────────────────
   checking for file ‘/private/var/folders/bg/pzsh15_96957bg6ylzt60tyc0000gp/T/R✔  checking for file ‘/private/var/folders/bg/pzsh15_96957bg6ylzt60tyc0000gp/T/RtmpXE3bKJ/file10f3d7616dbea/units/DESCRIPTION’
─  preparing ‘units’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  running ‘cleanup’
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘units_0.8-6.tar.gz’
   
Installing package into ‘/Users/tomasz/Library/R/arm64/4.4/library’
(as ‘lib’ is unspecified)
* installing *source* package ‘units’ ...
** using staged installation
configure: units: 0.8-6
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C++... yes
checking whether clang++ -arch arm64 -std=gnu++17 accepts -g... yes
checking for clang++ -arch arm64 -std=gnu++17 option to enable C++11 features... none needed
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for _Bool... no
checking for stdbool.h that conforms to C99 or later... yes
checking for error_at_line... no
checking for gcc... clang -arch arm64
checking whether the compiler supports GNU C... yes
checking whether clang -arch arm64 accepts -g... yes
checking for clang -arch arm64 option to enable C11 features... none needed
checking for XML_ParserCreate in -lexpat... yes
checking for udunits2.h... no
checking for udunits2/udunits2.h... no
checking for ut_read_xml in -ludunits2... no
configure: error: in '/private/var/folders/bg/pzsh15_96957bg6ylzt60tyc0000gp/T/Rtmpg7l1oF/R.INSTALL10f8e160a3137/units':
configure: error: 
--------------------------------------------------------------------------------
  Configuration failed because libudunits2.so was not found. Try installing:
    * deb: libudunits2-dev (Debian, Ubuntu, ...)
    * rpm: udunits2-devel (Fedora, EPEL, ...)
    * brew: udunits (OSX)
  If udunits2 is already installed in a non-standard location, use:
    --configure-args='--with-udunits2-lib=/usr/local/lib'
  if the library was not found, and/or:
    --configure-args='--with-udunits2-include=/usr/include/udunits2'
  if the header was not found, replacing paths with appropriate values.
  You can alternatively set UDUNITS2_INCLUDE and UDUNITS2_LIBS manually.
--------------------------------------------------------------------------------

See 'config.log' for more details
ERROR: configuration failed for package ‘units’
* removing ‘/Users/tomasz/Library/R/arm64/4.4/library/units’
* restoring previous ‘/Users/tomasz/Library/R/arm64/4.4/library/units’
Warning message:
In i.p(...) :
  installation of package ‘/var/folders/bg/pzsh15_96957bg6ylzt60tyc0000gp/T//RtmpXE3bKJ/file10f3d1ce6160b/units_0.8-6.tar.gz’ had non-zero exit status
tomasz@tomaszkalinows-WQVX units % find /opt/homebrew/lib | grep udunits
/opt/homebrew/lib/libudunits2.0.dylib
/opt/homebrew/lib/libudunits2.a
/opt/homebrew/lib/libudunits2.dylib
tomasz@tomaszkalinows-WQVX units % find /opt/homebrew | grep udunits 
/opt/homebrew/bin/udunits2
/opt/homebrew/include/udunits.h
/opt/homebrew/include/udunits2.h
/opt/homebrew/var/homebrew/linked/udunits
/opt/homebrew/lib/libudunits2.0.dylib
/opt/homebrew/lib/libudunits2.a
/opt/homebrew/lib/libudunits2.dylib
/opt/homebrew/opt/udunits
/opt/homebrew/Cellar/udunits
/opt/homebrew/Cellar/udunits/2.2.28
/opt/homebrew/Cellar/udunits/2.2.28/INSTALL_RECEIPT.json
/opt/homebrew/Cellar/udunits/2.2.28/bin
/opt/homebrew/Cellar/udunits/2.2.28/bin/udunits2
/opt/homebrew/Cellar/udunits/2.2.28/.brew
/opt/homebrew/Cellar/udunits/2.2.28/.brew/udunits.rb
/opt/homebrew/Cellar/udunits/2.2.28/include
/opt/homebrew/Cellar/udunits/2.2.28/include/udunits.h
/opt/homebrew/Cellar/udunits/2.2.28/include/converter.h
/opt/homebrew/Cellar/udunits/2.2.28/include/udunits2.h
/opt/homebrew/Cellar/udunits/2.2.28/sbom.spdx.json
/opt/homebrew/Cellar/udunits/2.2.28/README
/opt/homebrew/Cellar/udunits/2.2.28/lib
/opt/homebrew/Cellar/udunits/2.2.28/lib/libudunits2.0.dylib
/opt/homebrew/Cellar/udunits/2.2.28/lib/libudunits2.a
/opt/homebrew/Cellar/udunits/2.2.28/lib/libudunits2.dylib
/opt/homebrew/Cellar/udunits/2.2.28/COPYRIGHT
/opt/homebrew/Cellar/udunits/2.2.28/share
/opt/homebrew/Cellar/udunits/2.2.28/share/info
/opt/homebrew/Cellar/udunits/2.2.28/share/info/udunits2.info
/opt/homebrew/Cellar/udunits/2.2.28/share/info/udunits2prog.info
/opt/homebrew/Cellar/udunits/2.2.28/share/info/udunits2lib.info
/opt/homebrew/Cellar/udunits/2.2.28/share/doc
/opt/homebrew/Cellar/udunits/2.2.28/share/doc/udunits
/opt/homebrew/Cellar/udunits/2.2.28/share/doc/udunits/udunits2-prefixes.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/doc/udunits/udunits2.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/doc/udunits/CHANGE_LOG
/opt/homebrew/Cellar/udunits/2.2.28/share/doc/udunits/README
/opt/homebrew/Cellar/udunits/2.2.28/share/doc/udunits/udunits2-common.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/doc/udunits/udunits2-derived.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/doc/udunits/COPYRIGHT
/opt/homebrew/Cellar/udunits/2.2.28/share/doc/udunits/udunits2-accepted.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/doc/udunits/udunits2-base.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/udunits
/opt/homebrew/Cellar/udunits/2.2.28/share/udunits/udunits2-prefixes.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/udunits/udunits2.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/udunits/udunits2-common.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/udunits/udunits2-derived.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/udunits/udunits2-accepted.xml
/opt/homebrew/Cellar/udunits/2.2.28/share/udunits/udunits2-base.xml
/opt/homebrew/share/info/udunits2.info
/opt/homebrew/share/info/udunits2prog.info
/opt/homebrew/share/info/udunits2lib.info
/opt/homebrew/share/doc/udunits
/opt/homebrew/share/udunits
```

--- a/NEWS.md
+++ b/NEWS.md
@@ -20,6 +20,8 @@
 
 * Fix UBs in the C++ glue code; #380
 
+* Add support for `brew` path discovery in macOS; #384
+
 # version 0.8-5
 
 * avoid -Wformat-security warning on CRAN

--- a/configure
+++ b/configure
@@ -643,6 +643,7 @@ ac_header_cxx_list=
 ac_subst_vars='LTLIBOBJS
 UDUNITS2_DIR
 UD_CPPFLAGS
+BREW
 ac_ct_CC
 CFLAGS
 CC
@@ -4160,6 +4161,57 @@ ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ex
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
 
+# Extract the first word of "brew", so it can be a program name with args.
+set dummy brew; ac_word=$2
+{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
+printf %s "checking for $ac_word... " >&6; }
+if test ${ac_cv_path_BREW+y}
+then :
+  printf %s "(cached) " >&6
+else case e in #(
+  e) case $BREW in
+  [\\/]* | ?:[\\/]*)
+  ac_cv_path_BREW="$BREW" # Let the user override the test with a path.
+  ;;
+  *)
+  as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
+for as_dir in $PATH
+do
+  IFS=$as_save_IFS
+  case $as_dir in #(((
+    '') as_dir=./ ;;
+    */) ;;
+    *) as_dir=$as_dir/ ;;
+  esac
+    for ac_exec_ext in '' $ac_executable_extensions; do
+  if as_fn_executable_p "$as_dir$ac_word$ac_exec_ext"; then
+    ac_cv_path_BREW="$as_dir$ac_word$ac_exec_ext"
+    printf "%s\n" "$as_me:${as_lineno-$LINENO}: found $as_dir$ac_word$ac_exec_ext" >&5
+    break 2
+  fi
+done
+  done
+IFS=$as_save_IFS
+
+  ;;
+esac ;;
+esac
+fi
+BREW=$ac_cv_path_BREW
+if test -n "$BREW"; then
+  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $BREW" >&5
+printf "%s\n" "$BREW" >&6; }
+else
+  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
+printf "%s\n" "no" >&6; }
+fi
+
+
+if test  -n "${ac_cv_path_BREW}"  ; then
+    UD_CPPFLAGS="-I`${ac_cv_path_BREW} --prefix`/include"
+    LIBS="-L`${ac_cv_path_BREW} --prefix`/lib ${LIBS}"
+fi
+
 
 # Check whether --with-udunits2-include was given.
 if test ${with_udunits2_include+y}
@@ -4168,12 +4220,11 @@ then :
 fi
 
 if test  -n "$udunits2_include_path"  ; then
-   UD_CPPFLAGS="-I${udunits2_include_path}"
-else
-   if test  -n "${UDUNITS2_INCLUDE}"  ; then
-      UD_CPPFLAGS="-I${UDUNITS2_INCLUDE}"
-   fi
+    UD_CPPFLAGS="-I${udunits2_include_path} ${UD_CPPFLAGS}"
+elif test  -n "${UDUNITS2_INCLUDE}"  ; then
+    UD_CPPFLAGS="-I${UDUNITS2_INCLUDE} ${UD_CPPFLAGS}"
 fi
+CPPFLAGS="${UD_CPPFLAGS} ${CPPFLAGS}"
 
 
 # Check whether --with-udunits2-lib was given.
@@ -4183,11 +4234,9 @@ then :
 fi
 
 if test  -n "$udunits2_lib_path"  ; then
-   LIBS="-L${udunits2_lib_path} ${LIBS}"
-else
-   if test  -n "${UDUNITS2_LIBS}"  ; then
-      LIBS="-L${UDUNITS2_LIBS} ${LIBS}"
-   fi
+    LIBS="-L${udunits2_lib_path} ${LIBS}"
+elif test  -n "${UDUNITS2_LIBS}"  ; then
+    LIBS="-L${UDUNITS2_LIBS} ${LIBS}"
 fi
 
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for XML_ParserCreate in -lexpat" >&5
@@ -4237,8 +4286,6 @@ fi
 if test "${ac_cv_lib_expat_XML_ParserCreate}" = yes; then
    LIBS="${LIBS} -lexpat"
 fi
-
-CPPFLAGS="${UD_CPPFLAGS} ${CPPFLAGS}"
 
 ac_fn_cxx_check_header_compile "$LINENO" "udunits2.h" "ac_cv_header_udunits2_h" "$ac_includes_default"
 if test "x$ac_cv_header_udunits2_h" = xyes
@@ -4321,7 +4368,6 @@ as_fn_error $? "
 
 See 'config.log' for more details" "$LINENO" 5; } fi
 
-UD_CPPFLAGS="${UD_CPPFLAGS} ${CPPFLAGS}"
 
 
 

--- a/configure
+++ b/configure
@@ -4207,9 +4207,9 @@ printf "%s\n" "no" >&6; }
 fi
 
 
-if test  -n "${ac_cv_path_BREW}"  ; then
-    UD_CPPFLAGS="-I`${ac_cv_path_BREW} --prefix`/include"
-    LIBS="-L`${ac_cv_path_BREW} --prefix`/lib ${LIBS}"
+if test  -n "${BREW}"  ; then
+    UD_CPPFLAGS="-I`${BREW} --prefix`/include"
+    LIBS="-L`${BREW} --prefix`/lib ${LIBS}"
 fi
 
 

--- a/configure.ac
+++ b/configure.ac
@@ -20,36 +20,37 @@ AC_FUNC_ERROR_AT_LINE
 AC_PREREQ
 AC_PROG_CC
 
+AC_PATH_PROG([BREW], [brew])
+if test [ -n "${ac_cv_path_BREW}" ] ; then
+    UD_CPPFLAGS="-I`${ac_cv_path_BREW} --prefix`/include"
+    LIBS="-L`${ac_cv_path_BREW} --prefix`/lib ${LIBS}"
+fi
+
 AC_ARG_WITH([udunits2-include],
     AS_HELP_STRING([--with-udunits2-include=DIR],
 		[location of the udunits2 header files]),
 		[udunits2_include_path=$withval])
 if test [ -n "$udunits2_include_path" ] ; then
-   UD_CPPFLAGS="-I${udunits2_include_path}"
-else
-   if test [ -n "${UDUNITS2_INCLUDE}" ] ; then
-      UD_CPPFLAGS="-I${UDUNITS2_INCLUDE}"
-   fi
+    UD_CPPFLAGS="-I${udunits2_include_path} ${UD_CPPFLAGS}"
+elif test [ -n "${UDUNITS2_INCLUDE}" ] ; then
+    UD_CPPFLAGS="-I${UDUNITS2_INCLUDE} ${UD_CPPFLAGS}"
 fi
+CPPFLAGS="${UD_CPPFLAGS} ${CPPFLAGS}"
 
 AC_ARG_WITH([udunits2-lib],
     AS_HELP_STRING([--with-udunits2-lib=DIR],
 		[location of the udunits2 libraries]),
 		[udunits2_lib_path=$withval])
 if test [ -n "$udunits2_lib_path" ] ; then
-   LIBS="-L${udunits2_lib_path} ${LIBS}"
-else
-   if test [ -n "${UDUNITS2_LIBS}" ] ; then
-      LIBS="-L${UDUNITS2_LIBS} ${LIBS}"
-   fi
+    LIBS="-L${udunits2_lib_path} ${LIBS}"
+elif test [ -n "${UDUNITS2_LIBS}" ] ; then
+    LIBS="-L${UDUNITS2_LIBS} ${LIBS}"
 fi
 
 AC_CHECK_LIB(expat, XML_ParserCreate, [],[], ${LIBS})
 if test "${ac_cv_lib_expat_XML_ParserCreate}" = yes; then
    LIBS="${LIBS} -lexpat"
 fi
-
-CPPFLAGS="${UD_CPPFLAGS} ${CPPFLAGS}"
 
 AC_CHECK_HEADER(udunits2.h, UDUNITS2_DIR=0, [
   AC_CHECK_HEADER(udunits2/udunits2.h, UDUNITS2_DIR=1,
@@ -73,7 +74,6 @@ if test "${UD_ERROR}" != ""  ; then AC_MSG_FAILURE([
 --------------------------------------------------------------------------------
 ]) fi
 
-UD_CPPFLAGS="${UD_CPPFLAGS} ${CPPFLAGS}"
 AC_SUBST([LIBS])
 AC_SUBST([UD_CPPFLAGS])
 AC_SUBST([UDUNITS2_DIR])

--- a/configure.ac
+++ b/configure.ac
@@ -21,9 +21,9 @@ AC_PREREQ
 AC_PROG_CC
 
 AC_PATH_PROG([BREW], [brew])
-if test [ -n "${ac_cv_path_BREW}" ] ; then
-    UD_CPPFLAGS="-I`${ac_cv_path_BREW} --prefix`/include"
-    LIBS="-L`${ac_cv_path_BREW} --prefix`/lib ${LIBS}"
+if test [ -n "${BREW}" ] ; then
+    UD_CPPFLAGS="-I`${BREW} --prefix`/include"
+    LIBS="-L`${BREW} --prefix`/lib ${LIBS}"
 fi
 
 AC_ARG_WITH([udunits2-include],

